docs/memory-safety-continuum.md

```diff
@@ -0,0 +1,66 @@
+# Memory Safety Continuum
+
+At this SIG's January 25, 2024 meeting, we discussed the idea of a continuum when it comes to software memory safety.
+
+When discussing memory safety, it's easy to fall into the idea of a memory safety binary - either your software is "memory safe" or it is not. When considering reality - especially with re: to so much existing software (legacy or not) - it is more nuanced.
+
+This DRAFT document lays out the idea of the "Memory Safety Continuum". Eventually, this idea may be used as a foundation for other content which will help developers and organizations identify where they are on the continuum and how to get to where they want to be.
+
+## What is Memory Safety?
+
+Rather than using terms like "Memory Safe Language" and "Memory Unsafe Language", this SIG prefers the terms "memory safe by default" and "non-memory safe by default". Please see our [useful definitions](definitions.md) file for more information about memory safety and undefined behavior.
+
+## The Continuum
+
+This is a very rough idea of what the continuum might look like - from "least safety" to "most safety"
+
+* Using a non-memory safe by default language (such as C or C++) without any developer best practices or automated tooling to check for memory safety
+* Using a non-memory safe by default language with developer best practices, but no automated tooling to check for memory safety
+* Using a non-memory safe by default language with developer best practices and automated tooling to check for memory safety
+* Using a memory safe by default language (such as Rust, Go, Python, Java, JavaScript, C#) without developer best practices and automated tooling
+* Using a memory safe by default language with developer best practices
+* Using a memory safe by default language with developer best practices and automated tooling
```

Comment:
i wonder if we should consider cases such as:
* using memory safe by default language that depends upon unsafe language (i.e. through the use of external/foreign functions, or with Ptyhon extensions)
* using memory safe by default languages which runtime was built with/without developer best practices and automated tooling (an edge case, but one that may exist, specifically in OT/edge/proprietary devices)

Reply:
What about when you disable the memory safeties (in languages like Rust, Ada, and C#, where that's possible)? If there's a small amount (say, a few dozen lines or under 0.1%) that's probably fine. If you have a large package that disables safeties, it's really just a memory-unsafe language & should be treated that way.

If you formally *prove* a program has no memory safety issues (as opposed to using a typical SAST) tool, then I'd say you are using a memory-safe language (even if it's C). Of course, the number of programs (like seL4) where that true is small, but it *is* a possibility.

Reply:
Good points - I've added content about them in the FAQs.